### ISSUE TITLE
sketch-rnn: Time to train

### ISSUE THREAD

**Issue by alexis-jacq** [tensorflow/magenta#742]
Hello, thanks for this amazing algorithm!

I am writing a Pytorch version of sketch-rnn, and I would like to know what is the average time of training with a simple laptop's GPU. (using the same default hyperparameters, and only training to generate samples from the cat.npz dataset)

Also, what is approximatively the minimal number of epoch in order to get recognizable drawing generations?

**Comment by alexis-jacq** [comment]
Ok !
Thanks a lot for the tips !

**Comment by hardmaru** [comment]
You should be able to see some results (i.e. it should draw something that look like crabs if you crab), after 10-20K steps. We set the default number of steps close to infinity to allow us to control manually when we want to stop training, by looking at the TensorBoard graphs and the various reported losses. In practice once the validation loss curves stop decreasing we stop the training

The convergence depends on the data, and also the regularization scheme you use, and size of the network, so can vary around 50K-500K steps even.

**Comment by falaktheoptimist** [comment]
Ok. That helps a lot. Many thanks @hardmaru

**Comment by hardmaru** [comment]
The loss function for the decoder is somewhat non standard, so you may encounter some challenges if you want to implement in a high level framework such as Keras, but I think it should be possible.

If you manage to write the model in Keras in the future, please let us know so we can refer your implementation to Keras users or other interested parties.

**Comment by falaktheoptimist** [comment]
Ok sure.. We're making an attempt. Will share if we get some good results.

**Comment by alexiskattan** [comment]
@falaktheoptimist any luck getting sketch rnn to work in keras?

**Comment by KKeishiro** [comment]
@hardmaru @alexis-jacq @falaktheoptimist @alexiskattan 
I implemented Sketch-RNN in keras.
The repo can be found [here](https://github.com/KKeishiro/Sketch-RNN).
Any comments are welcomed. 
Cheers
